polaris-react/src/components/Text/Text.tsx

```diff
@@ -0,0 +1,83 @@
+import React, {ReactNode} from 'react';
+
+import {classNames} from '../../utilities/css';
+
+import styles from './Text.scss';
+
+type Element = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'p' | 'span';
+
+type Variant =
+  | 'headingSm'
+  | 'headingMd'
+  | 'headingLg'
+  | 'headingXl'
+  | 'heading2xl'
+  | 'heading3xl'
+  | 'heading4xl'
+  | 'bodySm'
+  | 'bodyMd'
+  | 'bodyLg';
+
+type Alignment = 'inherit' | 'start' | 'center' | 'end' | 'justify';
+
+type FontWeight = 'regular' | 'medium' | 'semibold' | 'bold';
+
+type Color = 'success' | 'critical' | 'warning' | 'subdued';
+
+const VariantFontWeightMapping: {[V in Variant]: FontWeight} = {
+  headingSm: 'bold',
+  headingMd: 'semibold',
+  headingLg: 'semibold',
+  headingXl: 'semibold',
+  heading2xl: 'semibold',
+  heading3xl: 'semibold',
+  heading4xl: 'bold',
+  bodySm: 'regular',
+  bodyMd: 'regular',
+  bodyLg: 'regular',
+};
+
+export interface TextProps {
+  /** Adjust horizontal alignment of text */
+  alignment?: Alignment;
+  /** The element type */
+  as: Element;
+  /** Text to display */
+  children: ReactNode;
+  /** Adjust color of text */
+  color?: Color;
+  /** Adjust weight of text */
+  fontWeight?: FontWeight;
+  /** Truncate text overflow with ellipsis */
+  truncate?: boolean;
+  /** Typographic style of text */
+  variant: Variant;
+  /** Visually hide the text */
+  visuallyHidden?: boolean;
+}
+
+export const Text = ({
+  alignment = 'inherit',
+  as,
+  children,
+  color,
+  fontWeight,
+  truncate = false,
+  variant,
+  visuallyHidden = false,
+}: TextProps) => {
+  const Component = as || (visuallyHidden ? 'span' : 'p');
+
+  const className = classNames(
+    styles.root,
+    styles[variant],
+    fontWeight ? styles[fontWeight] : styles[VariantFontWeightMapping[variant]],
+    (alignment || truncate) && styles.block,
```

Comment:
```suggestion
    (alignment || truncate) && styles.block,
```

I think this is making everything `block` by default. If the user has a `span` and `alignment = 'inherit'` this would make the element `display: block` which I don't think is desired. @chazdean

Reply:
I opened a PR [here](https://github.com/Shopify/polaris/pull/6797)

Reply:
Resolving this conversation, context in another PR [here](https://github.com/Shopify/polaris/pull/6804#discussion_r933425498)